core/src/main/java/bisq/core/api/CoreMoneroConnectionsService.java

```diff
@@ -283,6 +295,9 @@ private void initialize() {
             // run once
             if (!isInitialized) {
 
+                // starts a local in process node if user previously started
+                nodeService.startMoneroNode();
```

Comment:
This logic should be present: if the last used connection was local and it's currently offline, start the node.

nodeService.startMoneroNode() should always start the node or throw an error. It should use the last used settings, which default to blanks if never previously set.

Finally, your approach with the listener was correct imo: CoreMoneroConnectionService is added as listener on CoreMoneroNodeService. Then `onNodeStarted(daemon)` -> `setConnection(daemon.getRpcConnection())`, and `onNodeStopped()` -> `checkConnection()`

Reply:
Done, I check if the connection was local by matching against the loopback address. startMoneroNode now always attempts to start the local node, and the default value of the node settings object is valid (blanks).